robottelo/test.py

```diff
@@ -211,6 +232,128 @@ def setUp(self):
         self.user = User(self.browser)
         self.usergroup = UserGroup(self.browser)
 
+    def take_screenshot(self, testmethodname):
+        """Takes screenshot of the UI and saves it to the disk by creating
+        a directory same as that of the test method name."""
+        fmt = '%Y-%m-%d_%H:%M:%S'
+        filename = datetime.now().strftime(fmt)
+        # testmethodname varies so trying to handle various scenarios.
+        # pop depending on ddt, normal test or ddt's 1st job.
+        # For ddt tests with maps.
+        if "___" in testmethodname:
+            parts = testmethodname.split("___")
+        # For ddt tests with utf8, html, etc.
+        elif "__" in testmethodname:
+            parts = testmethodname.split("__")
+        # For ddt test with alpha, numeric, etc.
+        else:
+            testmethodname = testmethodname.split('_')
+            for _ in range(2):
+                testmethodname.pop()
+        if "___" in testmethodname or "__" in testmethodname:
+            testmethodname = parts[0]
+            testmethodname = testmethodname.split('_')
+            testmethodname.pop()
+        testmethodname = "_".join(testmethodname)
```

Comment:
Huh, while reading my earlier explanation, I cam across a bug, need to fix it. 
I think it will fail for **Non DDT* stuff currently.